client.go

```diff
@@ -38,6 +39,10 @@ var (
 	// defaultClient is used for performing requests without explicitly making
 	// a new client. It is purposely private to avoid modifications.
 	defaultClient = NewClient()
+
+	// We need to consume response bodied to maintain http connections, but
```

Comment:
It's just a trade-off for the time to drain the connection vs the time to setup a new one. The client should appear to behave the same either way, but limiting the reader prevent us from waiting for some very large, unexpected response when we could just as well drop the connection.